For what it's worth, I ran the script from FETCH_HEAD, so that may have something to do with it. I'll go grab the error for you though.

EDIT:
Traceback (most recent call last):
  File "lint.py", line 11, in <module>
    shell=True) or
  File "/usr/lib/python2.7/subprocess.py", line 544, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command 'git config branch.`git rev-parse --abbrev-ref HEAD`.merge' returned non-zero exit status 1

--- a/lint.py
+++ b/lint.py
@@ -1,16 +1,20 @@
 #!/usr/bin/env python
 
+from __future__ import print_function
+
 import sys
 import subprocess
 
 other_branch = None
 if len(sys.argv) > 1:
     other_branch = sys.argv[1]
-other_branch = (other_branch or
-                subprocess.check_output("git config branch.`git rev-parse --abbrev-ref HEAD`.merge",
-                                        shell=True) or
-                "develop"
-               )
+else:
+    try:
+        cmd = "git config branch.`git rev-parse --abbrev-ref HEAD`.merge"
+        other_branch = subprocess.check_output(cmd, shell=True)
+    except subprocess.CalledProcessError:
+        print("Defauting to linting against changes since develop", file=sys.stderr)
+        other_branch = "develop"
 
 subprocess.call("git diff `git merge-base HEAD %s` | flake8 --diff --show-source" % other_branch,
                 shell=True)

